src/Orleans.Runtime.Abstractions/Streams/ISiloPersistentStreamConfigurator.cs

```diff
@@ -0,0 +1,66 @@
+﻿using Orleans.Configuration;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Orleans.Streams
+{
+    public interface ISiloPersistentStreamConfigurator
```

Comment:
By not having some sort of final 'build' step this abstraction lacks a means of checking the completeness of the various required configuration steps for a specific provider.  This may be addressable with validators, or some other means, so I'm not pushing back on this pattern, just noting a characteristic of this abstraction.

Reply:
as we discussed offline, I'm hesitating to add that Validate or Builder method on the interface, since we have no way to force user to call that, and it is really the user's responsibility to call that. since we call validate in runtime. In short, it is not required for user to call it, then we shouldn't add it to the interface.